shared/src/simulation/behaviors/unload-arrived-vehicles.ts

```diff
@@ -0,0 +1,53 @@
+import { IsInt, IsUUID, Min } from 'class-validator';
+import { getCreate } from '../../models/utils';
+import { UUID, uuid, uuidValidationOptions } from '../../utils';
+import { IsValue } from '../../utils/validators';
+import { UnloadVehicleActivityState } from '../activities/unload-vehicle';
+import { addActivity } from '../utils/simulated-region';
+import type {
+    SimulationBehavior,
+    SimulationBehaviorState,
+} from './simulation-behavior';
+
+export class UnloadArrivedVehiclesBehaviorState
+    implements SimulationBehaviorState
+{
+    @IsValue('unloadVehiclesBehavior' as const)
+    readonly type = 'unloadVehiclesBehavior';
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly id: UUID = uuid();
+
+    @IsInt()
+    @Min(0)
+    public readonly unloadDelay!: number;
+
+    /**
+     * @deprecated Use {@link create} instead
+     */
+    constructor(unloadDelay: number) {
+        this.unloadDelay = unloadDelay;
+    }
+
+    static readonly create = getCreate(this);
+}
+
+export const unloadArrivingVehiclesBehavior: SimulationBehavior<UnloadArrivedVehiclesBehaviorState> =
+    {
+        behaviorState: UnloadArrivedVehiclesBehaviorState,
+        handleEvent(draftState, simulatedRegion, behaviorState, event) {
+            if (
+                event.type === 'vehicleArrivedEvent' &&
+                event.simulatedRegionId === simulatedRegion.id
```

Comment:
I think this check (and therefore the `simulatedRegionId` property on the event at all) is not necessary since `handleEvents` will only be called for events related to the simulated region passed as argument.